app/uk/gov/hmrc/gform/models/FieldValueRaw.scala

```diff
@@ -65,7 +65,7 @@ case class FieldValueRaw(
 
   def toFieldValue = Reads[FieldValue] { _ => getFieldValue fold (us => JsError(us.toString), fv => JsSuccess(fv)) }
 
-  private def getFieldValue(): Opt[FieldValue] = optMES.flatMap(mes => optComponentType.map(ct => mkFieldValue(mes, ct)))
+  private def getFieldValue(): Either[UnexpectedState, FieldValue] = optMES.flatMap(mes => fromRawToFinishedComponentType.map(ct => mkFieldValue(mes, ct)))
```

Comment:
I know that it's pain for the first time when you see it but once you remember it it saves much more time. Especially that it's widely used in this project :)